src/nvim/ops.c

```diff
@@ -1686,9 +1686,10 @@ int op_replace(oparg_T *oap, int c)
 
       oldp = get_cursor_line_ptr();
       oldlen = STRLEN(oldp);
-      assert(n >= 0);
-      newp = (char_u *)xmalloc(oldlen + 1 + (size_t)n);
-      memset(newp, NUL, oldlen + 1 + (size_t)n);
+      int new_size = (int)oldlen + 1 + n;
+      assert(new_size >= 0 && oldlen <= (size_t)(INT_MAX - 1 - n));
```

Comment:
If there was an overflow in the computation of the initial value of `new_size`, the overflow check here is too late.